### PR TITLE
win32_hooks refactoring and unit tests

### PR DESCRIPTION
**Pull request by airelil**
win32_hooks: 
- refactor events loop, low-level handlers
- fix warnings on return from "finally"
- more constants in ID_TO_KEY dictionary
- add a test suite for win32_hooks
  - test with real keys/mouse through SendKeys/mouse_click
  - test with mocking system dependencies

win32defines: 
- group and sort VK_ and WM_ constants

### CODE REVIEW

File: pywinauto/win32_hooks.py [hunk 1]
```diff
@@ -241,83 +407,143 @@ class Hook(object):
...
-                    event_type = self.event_types[0xFFFFFFFF & event_code]
-
-                    if event_type == 'key down':
-                        self.pressed_keys.append(current_key)
-
-                    if event_type == 'key up':
-                        self.pressed_keys.remove(current_key)
-
-                    event = KeyboardEvent(current_key, event_type, self.pressed_keys)
+            @HOOKCB
```
Comment (vasily-v-ryabov) [inline pywinauto/win32_hooks.py]: Wow! Didn't know `CFUNCTYPE` object can be used as a decorator. 👍

File: pywinauto/win32defines.py [hunk 2]
```diff
@@ -34,23 +34,21 @@
 
 """Definition of Windows defines"""
 from __future__ import unicode_literals
+from ctypes import c_int
 
 __revision__ = "$Revision$"
```
Comment (vasily-v-ryabov) [inline pywinauto/win32defines.py]: I think these `__revision__` lines could be removed. It's inherited from Mercurial or SVN probably.

File: pywinauto/win32_hooks.py [hunk 3]
```diff
@@ -241,83 +407,143 @@ class Hook(object):
...
+                al = ActionLogger()
+                al.log("_process_kbd_msg_type, can't remove a key: {0}".format(current_key))
+
+        return event_type
+
+    def _keyboard_ll_hdl(self, code, event_code, kb_data_ptr):
+        """Execute when a keyboard low level event was catched"""
+        try:
+            # The next hook in chain must be always called
+            res = windll.user32.CallNextHookEx(self.keyboard_id,
```
Comment (airelil) [inline pywinauto/win32_hooks.py]: The MSDN docs say: "A hook procedure can call this function either before or after processing the hook information." Notice that the hooks run in the context of an application so, I think, the ```CallNextHookEx``` simply sends a message to another registered app (we by ourselves have the loop message) Maybe it also calls to ```LoadLibrary```.

**Comment by airelil** [comment]
Regarding pywin32 dependency. It's an interesting point. We don't see any users complaining about a burden of this dependency and we already have additional one - comtypes. I'd remove it only if the package is stopped be actively maintained or if it prevents from migrating to Python3

**Comment by vasily-v-ryabov** [comment]
Great! Now everything looks fine. Thanks a lot!

**Pull request merged**